packages/amplify-cli/src/lib/init-steps/preInitSetup.js

```diff
@@ -0,0 +1,87 @@
+const { execSync } = require('child_process');
+const { getPackageManager } = require('../packageManagerHelpers');
+const { normalizePackageManagerForOS } = require('../packageManagerHelpers');
+const { generateLocalEnvInfoFile } = require('./s9-onSuccess');
+const url = require('url');
+
+async function run(context) {
+  if (context.parameters.options.app) {
+    // Setting up a sample app
+    context.print.warning('Note: Amplify does not have knowledge of the url provided');
+    const repoUrl = context.parameters.options.app;
+    await validateGithubRepo(context, repoUrl);
+    await cloneRepo(context, repoUrl);
+    await installPackage();
+    await setLocalEnvDefaults(context);
+  }
+  return context;
+}
+
+/**
+ * Checks whether a url is a valid remote github repository
+ *
+ * @param repoUrl the url to validated
+ * @throws error if url is not a valid remote github url
+ */
+async function validateGithubRepo(context, repoUrl) {
+  try {
+    url.parse(repoUrl);
+    execSync(`git ls-remote ${repoUrl}`, { stdio: 'ignore' });
+  } catch (e) {
+    context.print.error('Invalid remote github url');
+    process.exit(1);
+  }
+}
+
+/**
+ * Clones repo from url to current directory (must be empty)
+ *
+ * @param repoUrl the url to be cloned
+ */
+async function cloneRepo(context, repoUrl) {
+  try {
+    execSync(`git clone ${repoUrl} .`, { stdio: 'inherit' });
+  } catch (e) {
+    context.print.error('Please ensure you run this command in an empty directory');
+    process.exit(1);
+  }
+}
+
+/**
+ * Install package using the correct package manager if package handling file exists
+ *
+ * @param packageManager either npm or yarn
+ */
+async function installPackage() {
+  const packageManager = await getPackageManager();
+  const normalizedPackageManager = await normalizePackageManagerForOS(packageManager);
+  if (packageManager === 'yarn') {
```

Comment:
do we need this condition? In both the cases arguments for `execSync ` to be the same

Reply:
Removing since its redundant